dev/src/expression.ts

```diff
@@ -0,0 +1,1425 @@
+  array: Expr | string,
+  elements: any[]
+): ArrayConcat {
+  const arrayExpr = array instanceof Expr ? array : Field.of(array);
+  const exprValues = elements.map(element =>
+    element instanceof Expr ? element : Constant.of(element)
+  );
+  return new ArrayConcat(arrayExpr, exprValues);
+}
+
+export function arrayContains(array: Expr, element: Expr): ArrayContains;
+export function arrayContains(array: Expr, element: any): ArrayContains;
+export function arrayContains(array: string, element: Expr): ArrayContains;
+export function arrayContains(array: string, element: any): ArrayContains;
+export function arrayContains(
+  array: Expr | string,
+  element: any
+): ArrayContains {
+  const arrayExpr = array instanceof Expr ? array : Field.of(array);
+  const elementExpr = element instanceof Expr ? element : Constant.of(element);
+  return new ArrayContains(arrayExpr, elementExpr);
+}
+
+export function arrayContainsAny(array: Expr, values: Expr[]): ArrayContainsAny;
+export function arrayContainsAny(array: Expr, values: any[]): ArrayContainsAny;
+export function arrayContainsAny(
+  array: string,
+  values: Expr[]
+): ArrayContainsAny;
+export function arrayContainsAny(
+  array: string,
+  values: any[]
+): ArrayContainsAny;
+export function arrayContainsAny(
+  array: Expr | string,
+  values: any[]
+): ArrayContainsAny {
+  const arrayExpr = array instanceof Expr ? array : Field.of(array);
+  const exprValues = values.map(value =>
+    value instanceof Expr ? value : Constant.of(value)
+  );
+  return new ArrayContainsAny(arrayExpr, exprValues);
+}
+
+export function arrayContainsAll(array: Expr, values: Expr[]): ArrayContainsAll;
+export function arrayContainsAll(array: Expr, values: any[]): ArrayContainsAll;
+export function arrayContainsAll(
+  array: string,
+  values: Expr[]
+): ArrayContainsAll;
+export function arrayContainsAll(
+  array: string,
+  values: any[]
+): ArrayContainsAll;
+export function arrayContainsAll(
+  array: Expr | string,
+  values: any[]
+): ArrayContainsAll {
+  const arrayExpr = array instanceof Expr ? array : Field.of(array);
+  const exprValues = values.map(value =>
+    value instanceof Expr ? value : Constant.of(value)
+  );
+  return new ArrayContainsAll(arrayExpr, exprValues);
+}
+
+export function arrayFilter(array: Expr, filter: FilterExpr): ArrayFilter {
+  return new ArrayFilter(array, filter);
+}
+
+export function arrayLength(array: Expr): ArrayLength {
+  return new ArrayLength(array);
+}
+
+export function arrayTransform(
+  array: Expr,
+  transform: Function
+): ArrayTransform {
+  return new ArrayTransform(array, transform);
+}
+
+export function arrayElement(): ArrayElement {
+  return new ArrayElement();
+}
+
+export function inAny(element: Expr, others: Expr[]): In;
+export function inAny(element: Expr, others: any[]): In;
+export function inAny(element: string, others: Expr[]): In; // Added overload
+export function inAny(element: string, others: any[]): In; // Added overload
```

Comment:
If this behavior is different from the classic `in()` supporting array, should we also change `Expr.in(...)` to name `Expr.inAny(...)`. Or if this is only supporting `in` arrays, would it make more sense to name it `inArray(...)`?

Reply:
Will rename Expr.in to inAny.